lib/AST/ASTContext.cpp

```diff
@@ -908,6 +914,75 @@ FuncDecl *ASTContext::getEqualIntDecl() const {
   return nullptr;
 }
 
+FuncDecl *ASTContext::getMutatingXorIntDecl() const {
```

Comment:
Thinking out loud: This code has been copied and pasted enough times it's probably worth it to factor it out at some point...